demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1354 @@
+                   AWS_IOT_ENDPOINT ) );
+
+        /* Sends an MQTT Connect packet using the established TLS session,
+         * then waits for connection acknowledgment (CONNACK) packet. */
+        returnStatus = establishMqttSession( &mqttContext );
+
+        if( returnStatus != EXIT_SUCCESS )
+        {
+            LogError( ( "Failed creating an MQTT connection to %.*s.",
+                        AWS_IOT_ENDPOINT_LENGTH,
+                        AWS_IOT_ENDPOINT ) );
+        }
+        else
+        {
+            LogDebug( ( "Success creating MQTT connection to %.*s.",
+                        AWS_IOT_ENDPOINT_LENGTH,
+                        AWS_IOT_ENDPOINT ) );
+
+            mqttSessionEstablished = true;
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static void disconnect( void )
+{
+    /* Disconnect from broker. */
+    LogInfo( ( "Disconnecting the MQTT connection with %s.", AWS_IOT_ENDPOINT ) );
+
+    MQTT_Disconnect( &mqttContext );
+
+    /* End TLS session, then close TCP connection. */
+    ( void ) Openssl_Disconnect( &networkContext );
+}
+
+/*-----------------------------------------------------------*/
+
+static OtaMqttStatus_t mqttSubscribe( const char * pTopicFilter,
+                                      uint16_t topicFilterLength,
+                                      uint8_t qos,
+                                      OtaMqttCallback_t callback )
+{
+    OtaMqttStatus_t otaRet = OtaMqttSuccess;
+
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTContext_t * pMqttContext = &mqttContext;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    assert( pMqttContext != NULL );
+    assert( pTopicFilter != NULL );
+    assert( topicFilterLength > 0 );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* Set the QoS , topic and topic length. */
+    pSubscriptionList[ 0 ].qos = qos;
+    pSubscriptionList[ 0 ].pTopicFilter = pTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = topicFilterLength;
+
+    /* Send SUBSCRIBE packet. */
+    mqttStatus = MQTT_Subscribe( pMqttContext,
+                                 pSubscriptionList,
+                                 sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                 MQTT_GetPacketId( pMqttContext ) );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send SUBSCRIBE packet to broker with error = %u.",
+                    mqttStatus ) );
+
+        otaRet = OtaMqttSubscribeFailed;
+    }
+    else
+    {
+        LogInfo( ( "SUBSCRIBE topic %.*s to broker.\n\n",
+                   topicFilterLength,
+                   pTopicFilter ) );
+    }
+
+    /* Register callback to subscription manager. */
+    SubscriptionManager_RegisterCallback( pTopicFilter, topicFilterLength, callback );
```

Comment:
The return status of the subscription manager should be handled

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/6fa7ad842095edb46f52d9d38a6e27ffa26a21eb